File: java/src/main/java/com/allstask/api/Connect.java
// this file is @generated
package com.allstask.api;

import com.allstask.AllstaskHttpClient;
import com.allstask.exceptions.ApiException;
import com.allstask.models.ConnectedAccount;
import com.allstask.models.ConnectedAccountsResponse;
import com.allstask.models.CreateConnectedAccountRequest;
import com.allstask.models.CreateOnboardingLinkRequest;
import com.allstask.models.OnboardingLinkResponse;

import okhttp3.HttpUrl;

import java.io.IOException;

public class Connect {
    private final AllstaskHttpClient client;

    public Connect(AllstaskHttpClient client) {
        this.client = client;
    }

    /** List all connected accounts for this platform. */
    public ConnectedAccountsResponse listConnectedAccounts() throws IOException, ApiException {

        HttpUrl.Builder url = this.client.newUrlBuilder().encodedPath("/api/v1/connected-accounts");
        return this.client.executeRequest(
                "GET", url.build(), null, null, ConnectedAccountsResponse.class);
    }

    /**
     * Create a new connected account (Express flow). Returns the account and an onboarding link for
     * the user to complete setup.
     */
    public ConnectedAccount createConnectedAccount(
            final CreateConnectedAccountRequest createConnectedAccountRequest)
            throws IOException, ApiException {
        HttpUrl.Builder url = this.client.newUrlBuilder().encodedPath("/api/v1/connected-accounts");
        return this.client.executeRequest(
                "POST", url.build(), null, createConnectedAccountRequest, ConnectedAccount.class);
    }

    /** Retrieve a connected account by ID. */
    public ConnectedAccount getConnectedAccount(final String id) throws IOException, ApiException {
        HttpUrl.Builder url =
                this.client
                        .newUrlBuilder()
                        .encodedPath(String.format("/api/v1/connected-accounts/%s", id));
        return this.client.executeRequest("GET", url.build(), null, null, ConnectedAccount.class);
    }

    /** Revoke a connected account. All associated tokens are invalidated. */
    public void disconnectAccount(final String id) throws IOException, ApiException {
        HttpUrl.Builder url =
                this.client
                        .newUrlBuilder()
                        .encodedPath(String.format("/api/v1/connected-accounts/%s", id));
        this.client.executeRequest("DELETE", url.build(), null, null, null);
    }

    /**
     * Generate a new onboarding link for a connected account. Any existing unused link is
     * invalidated. The link expires after a configured duration.
     */
    public OnboardingLinkResponse createOnboardingLink(
            final String id, final CreateOnboardingLinkRequest createOnboardingLinkRequest)
            throws IOException, ApiException {
        HttpUrl.Builder url =
                this.client
                        .newUrlBuilder()
                        .encodedPath(String.format("/api/v1/connected-accounts/%s/onboarding", id));
        return this.client.executeRequest(
                "POST",
                url.build(),
                null,
                createOnboardingLinkRequest,
                OnboardingLinkResponse.class);
    }
}

File: typescript/src/api/connect.ts
// this file is @generated
import {
    ConnectedAccount,
    ConnectedAccountSerializer,
} from '../models/connectedAccount';
import {
    ConnectedAccountsResponse,
    ConnectedAccountsResponseSerializer,
} from '../models/connectedAccountsResponse';
import {
    CreateConnectedAccountRequest,
    CreateConnectedAccountRequestSerializer,
} from '../models/createConnectedAccountRequest';
import {
    CreateOnboardingLinkRequest,
    CreateOnboardingLinkRequestSerializer,
} from '../models/createOnboardingLinkRequest';
import {
    OnboardingLinkResponse,
    OnboardingLinkResponseSerializer,
} from '../models/onboardingLinkResponse';
import { HttpMethod, AllstaskRequest, AllstaskRequestContext } from "../request";

export class Connect {
    public constructor(private readonly requestCtx: AllstaskRequestContext) {}

    /** List all connected accounts for this platform. */
        public listConnectedAccounts(
            ): Promise<ConnectedAccountsResponse> {
            const request = new AllstaskRequest(HttpMethod.GET, "/api/v1/connected-accounts");

            
                return request.send(
                    this.requestCtx,
                    ConnectedAccountsResponseSerializer._fromJsonObject,
                );
            }

        

    /**
* Create a new connected account (Express flow). Returns the account
* and an onboarding link for the user to complete setup.
*/
        public createConnectedAccount(
            createConnectedAccountRequest: CreateConnectedAccountRequest,
            ): Promise<ConnectedAccount> {
            const request = new AllstaskRequest(HttpMethod.POST, "/api/v1/connected-accounts");

            request.setBody(
                    CreateConnectedAccountRequestSerializer._toJsonObject(
                        createConnectedAccountRequest,
                    )
                );
            
                return request.send(
                    this.requestCtx,
                    ConnectedAccountSerializer._fromJsonObject,
                );
            }

        

    /** Retrieve a connected account by ID. */
        public getConnectedAccount(
            id: string,
            ): Promise<ConnectedAccount> {
            const request = new AllstaskRequest(HttpMethod.GET, "/api/v1/connected-accounts/{id}");

            request.setPathParam("id", id);
            
                return request.send(
                    this.requestCtx,
                    ConnectedAccountSerializer._fromJsonObject,
                );
            }

        

    /** Revoke a connected account. All associated tokens are invalidated. */
        public disconnectAccount(
            id: string,
            ): Promise<void> {
            const request = new AllstaskRequest(HttpMethod.DELETE, "/api/v1/connected-accounts/{id}");

            request.setPathParam("id", id);
            
                return request.sendNoResponseBody(this.requestCtx);
            }

        

    /**
* Generate a new onboarding link for a connected account. Any existing
* unused link is invalidated. The link expires after a configured duration.
*/
        public createOnboardingLink(
            id: string,
            createOnboardingLinkRequest: CreateOnboardingLinkRequest,
            ): Promise<OnboardingLinkResponse> {
            const request = new AllstaskRequest(HttpMethod.POST, "/api/v1/connected-accounts/{id}/onboarding");

            request.setPathParam("id", id);
            request.setBody(
                    CreateOnboardingLinkRequestSerializer._toJsonObject(
                        createOnboardingLinkRequest,
                    )
                );
            
                return request.send(
                    this.requestCtx,
                    OnboardingLinkResponseSerializer._fromJsonObject,
                );
            }

        

    }

